src/utils/word-api.ts

```diff
@@ -2,12 +2,19 @@
 import { StatefulCitation } from "citeproc";
 
 class WordApi {
-  static async insertEmptyContentControl(): Promise<unknown> {
+  async insertNewCitation(
+    citations: Array<[number, string, StatefulCitation]>
+  ): Promise<unknown> {
     return Word.run((context: Word.RequestContext) => {
-      const getSelection = context.document.getSelection();
-      const contentControl = getSelection.insertContentControl();
-      contentControl.tag = "JABREF-CITATION-NEW";
-      contentControl.appearance = "BoundingBox";
+      const citationContentControl = context.document
+        .getSelection()
+        .insertContentControl();
+      citations.forEach((citation) => {
+        const citationText = citation[1];
+        citationContentControl.appearance = "BoundingBox";
+        citationContentControl.insertText(citationText, "Replace");
+        citationContentControl.tag = this.generateCitationTag(citation[2]);
```

Comment:
This code feels strange, or maybe I'm missing something. If you pass multiple citations, you are overwriting the text and tag in the next iteration. So what you end up with is essentially only the last citation, or not?